Styles/feed view

jphorner opened the pull request:
### Features
- Added containers for page components to enable consistent orientation
- Changed main font for app
- Added background color to sentiment options
- Added hover states to sentiment options
- Adjusted width of component displays
- Added borders and spacing to JSX elements

### Refactors 
- Removed form container as the other CSS rules accomplished the same goal
- Added comments within CSS files to explain styling decisions

### Issues and Technical Debt
- 

### Other Comments, Concerns, or Questions
- 

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature work
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran Lighthouse and WAVE accessibility audits and fixed errors/warnings
- [ ] Any dependent changes have been merged and published in downstream modules

jphorner commented on src/components/App/App.css:
```diff
@@ -1,10 +1,27 @@
+  height: 100%;
+  font-family: 'Comfortaa';
+}
+
```
Here, HTML rules are applied to ensure there is no white space when components are re-sized.

jphorner commented on src/components/Card/Card.css:
```diff
@@ -2,3 +2,32 @@
+  flex-direction: row;
+  justify-content: center;
+}
+
```
The card container has no direct styling and only serves to orient the contents
of each card. As explained in Feed.css, applying flex-row to the container
prevents any context issues caused by the native centering in React.

jphorner commented on src/components/Feed/Feed.css:
```diff
@@ -0,0 +1,15 @@
+  flex-direction: row;
+  justify-content: center;
+}
+
```
The article container is set to flex-row to keep the feed centered on the page.
This means that flex-column can be applied to the articles display while
keeping the row-centered orientation.

React natively has some center orientation applied, but it can cause issues when
orienting individual elements. Applying flex-center to a container overrides the
native centering in the cascade and makes it easier to troubleshoot styling
issues.

najinl replied:
I think we might need to add a CSS property for the article-image class if we are going to go the route of phone application, as when I set my dev tools to iPhone 6/7/8 or Galaxy S5, the article images start to overflow the articles-container.

najinl replied:
I think we might need to add a CSS property for the article-image class if we are going to go the route of phone application, as when I set my dev tools to iPhone 6/7/8 or Galaxy S5, the article images start to overflow the articles-container.
![Screen Shot 2021-11-01 at 11 29 04 AM](https://user-images.githubusercontent.com/68795578/139714086-10a336fe-2047-477d-8a7f-85ced898c625.png)

jphorner replied:
Got it! I'll start adding media queries to resolve this during my refactor. I noticed that happening this morning, too.

jphorner commented on src/components/Form/Form.css:
```diff
@@ -18,21 +19,41 @@ article {
+  width: 50%;
+  background-color: #42C7E4;
+}
+
```
Here, .sentiments-container holds the buttons while .sentiment-selection serves
  to display the light blue stripe and keep .sentiments-container centered.

  To make this work, .sentiment-selection and .sentiments-container both have
  flex applied independently. This is to avoid spacing the buttons via margins
  or padding, which would cause them to shift position when the hover state
  is activated.

  To summarize, .sentiment-selection has a width of 100% to hold the blue stripe and
  to center the container holding the buttons, and .sentiments-container is set to
  50% width + space-around to match the other page elements and enable non-
  intrusive hover states.

nzambonivergara reviewed (changes requested):
Great job, I really like the styling of our application! I left one minor comment, and I am attaching a screenshot of how the buttons look when I view our application for mobile devices using the dev tools. I am not entirely sure what is happening there, or if it is just in my device, but maybe something we can look into together?
![Screen Shot 2021-11-01 at 10 57 52 AM](https://user-images.githubusercontent.com/83611094/139711946-cd3ebaff-b682-4742-ad89-a47ce31b024e.png)

jphorner commented on src/components/App/App.css:
```diff
@@ -1,10 +1,29 @@
+html {
```
Thanks for the review! Yes, I can move the html rule into index.css, and I'll apply some media queries for screen re-sizing. (I set the width of the elements to 50% a bit arbitrarily. I'll be sure to use defined mobile screen sizes from now on.)

I'll do that today and hopefully we'll get it resolved for a merge soon.

jphorner replied:
Good call. I'll be sure to move it.

jphorner commented on src/components/App/App.css:
```diff
@@ -1,10 +1,27 @@
 .App-header {
-  background-color: #282c34;
-  min-height: 20vh;
+  /* background-color: #282c34; */
```
Whoops - will do! My mistake.

nzambonivergara commented on src/components/Form/Form.tsx:
```diff
@@ -13,23 +13,23 @@ const Form = ({ changeUserSentiment } : FormProps) : JSX.Element => {
-    </div>
+        <div className="sentiment-selection">
+          <div className="sentiments-container">
+            <button className="happy-btn" name="1" onClick={assignSentiment}></button>
```
The accessibility audits are flagging the fact that our buttons are empty. The solution WAVE provides: "Place text content within the <button> element or give the <input> element a value attribute." Just making a note for us to discuss how to address this.

jphorner replied:
Good catch. I think assigning a value of 'happy', etc. would be the best option to avoid interference with the image itself. If that sounds good enough, I can add 'happy', 'neutral', and 'sad' values during my refactor this afternoon.

jphorner replied:
(GH deleted my response for some reason? Let me try again.)

Good catch - I think assigning values of 'happy', 'neutral', and 'sad' would be the best option since we don't want text on the buttons. I can add those during my refactor this afternoon if that sounds good to you!

najinl replied:
This is something that should be on my docket to update. I'm about to push my changes with router and will update this when I'm in there!

jphorner replied:
Perfect! Thank you, Nadia. I'll just focus on the CSS changes for now to avoid merge conflicts. :)

najinl commented:
Thank you for make the app BEAUTIFUL! Appreciate you.

jphorner commented:
**Update:**

- Added media queries to cover all screen sizes
- Moved background styling from App.css to index.css
- Removed commented-out rules

**To be changed later on:**

- Resizing still feels a bit jittery
- Inconsistent margin ratios; may need to use pixel units
- More consistent image sizing


All that said, there should be no major issues when viewing from multiple screen sizes. Let me know what you think!
